TestRedirectRevalidation urllib3 tests fail when urllib3 is not installed

**Describe the bug**

Since #1668 (commit 919d827), CI on `main` has been red. The four `test_urllib3_*` tests in `xrspatial/geotiff/tests/test_ssrf_hardening_1664.py::TestRedirectRevalidation` raise `ModuleNotFoundError: No module named 'urllib3'` on every OS / Python combo:

\`\`\`
FAILED xrspatial/geotiff/tests/test_ssrf_hardening_1664.py::TestRedirectRevalidation::test_urllib3_redirect_to_private_rejected
FAILED xrspatial/geotiff/tests/test_ssrf_hardening_1664.py::TestRedirectRevalidation::test_urllib3_redirect_to_public_followed
FAILED xrspatial/geotiff/tests/test_ssrf_hardening_1664.py::TestRedirectRevalidation::test_urllib3_redirect_chain_capped
FAILED xrspatial/geotiff/tests/test_ssrf_hardening_1664.py::TestRedirectRevalidation::test_urllib3_relative_location_resolved
===== 4 failed, 4683 passed, 999 skipped =====
\`\`\`

`urllib3` is an optional runtime dependency (`_HTTPSource` falls back to stdlib `urllib.request` when urllib3 is missing — see `_reader.py:615-617`). The test class mocks urllib3's `PoolManager` API and calls `src.read_range(...)`, which calls `_urllib3_timeout()` at `_reader.py:640`. That helper does a lazy `import urllib3` and blows up when the package isn't installed.

**Expected behavior**

The four `test_urllib3_*` tests are specifically exercising the urllib3 path. They should skip cleanly when urllib3 is not installed.

**Suggested fix**

Add `pytest.importorskip("urllib3")` at the top of the `TestRedirectRevalidation` class. The stdlib-fallback path can be tested separately if needed.

**Impact**

Currently blocks CI on every PR that branches from `main`. PRs #1675-#1680 all hit this same wall.

--- a/xrspatial/geotiff/tests/test_ssrf_hardening_1664.py
+++ b/xrspatial/geotiff/tests/test_ssrf_hardening_1664.py
@@ -267,18 +267,17 @@ class _MockPool:
 
 
 class TestRedirectRevalidation:
-    # The tests in this class exercise the urllib3 transport path: they mock
+    # The test_urllib3_* tests exercise the urllib3 transport path: they mock
     # urllib3.PoolManager and call read_range(), which internally builds a
     # urllib3.Timeout via _urllib3_timeout(). urllib3 is an optional runtime
     # dependency (_HTTPSource falls back to stdlib urllib.request when it's
-    # missing -- see _reader.py:615-617). Skip rather than fail when the
-    # package isn't installed; the stdlib fallback path is covered elsewhere.
-    @pytest.fixture(autouse=True)
-    def _require_urllib3(self):
-        pytest.importorskip("urllib3")
+    # missing -- see _reader.py:615-617), so each urllib3-using test starts
+    # with pytest.importorskip("urllib3"). The test_stdlib_* tests below
+    # exercise the stdlib redirect handler directly and run regardless.
 
     def test_urllib3_redirect_to_private_rejected(self, monkeypatch):
         """Public host that 302-redirects to loopback must be rejected."""
+        pytest.importorskip("urllib3")
         # Initial validator pass: example.com resolves to a public IP.
         monkeypatch.setattr(
             socket, 'getaddrinfo', _fake_getaddrinfo('93.184.216.34'))
@@ -300,6 +299,7 @@ class TestRedirectRevalidation:
 
     def test_urllib3_redirect_to_public_followed(self, monkeypatch):
         """Public -> public redirect is followed; validator passes each hop."""
+        pytest.importorskip("urllib3")
         monkeypatch.setattr(
             socket, 'getaddrinfo', _fake_getaddrinfo('93.184.216.34'))
         src = _reader_mod._HTTPSource('https://example.com/cog.tif')
@@ -315,6 +315,7 @@ class TestRedirectRevalidation:
 
     def test_urllib3_redirect_chain_capped(self, monkeypatch):
         """More than _HTTP_MAX_REDIRECTS hops raises rather than looping."""
+        pytest.importorskip("urllib3")
         monkeypatch.setattr(
             socket, 'getaddrinfo', _fake_getaddrinfo('93.184.216.34'))
         src = _reader_mod._HTTPSource('https://example.com/cog.tif')
@@ -332,6 +333,7 @@ class TestRedirectRevalidation:
 
     def test_urllib3_relative_location_resolved(self, monkeypatch):
         """Relative Location like ``/other.tif`` resolves against the source."""
+        pytest.importorskip("urllib3")
         monkeypatch.setattr(
             socket, 'getaddrinfo', _fake_getaddrinfo('93.184.216.34'))
         src = _reader_mod._HTTPSource('https://example.com/dir/cog.tif')

--- a/xrspatial/geotiff/tests/test_ssrf_hardening_1664.py
+++ b/xrspatial/geotiff/tests/test_ssrf_hardening_1664.py
@@ -267,6 +267,16 @@ class _MockPool:
 
 
 class TestRedirectRevalidation:
+    # The tests in this class exercise the urllib3 transport path: they mock
+    # urllib3.PoolManager and call read_range(), which internally builds a
+    # urllib3.Timeout via _urllib3_timeout(). urllib3 is an optional runtime
+    # dependency (_HTTPSource falls back to stdlib urllib.request when it's
+    # missing -- see _reader.py:615-617). Skip rather than fail when the
+    # package isn't installed; the stdlib fallback path is covered elsewhere.
+    @pytest.fixture(autouse=True)
+    def _require_urllib3(self):
+        pytest.importorskip("urllib3")
+
     def test_urllib3_redirect_to_private_rejected(self, monkeypatch):
         """Public host that 302-redirects to loopback must be rejected."""
         # Initial validator pass: example.com resolves to a public IP.

